src/main/java/ui/BoardAutoCreator.java

```diff
@@ -74,6 +87,79 @@ private void createMilestoneBoard() {
                         "It is saved under the name \"" + boardName + "\".");
     }
 
+    private void createWorkAllocationBoard() {
+        logger.info("Creating " + WORK_ALLOCATION + " board");
+
+        List<TurboUser> userList = ui.logic.getRepo(ui.logic.getDefaultRepo()).getUsers();
+
+        if (userList.isEmpty()) {
+            DialogMessage.showInformationDialog("Auto-create Board - " + WORK_ALLOCATION,
+                    WORK_ALLOCATION + " board cannot be created as your" +
+                            " account has no push access to the default repo.");
+
+            return;
+        }
+
+        panelControl.closeAllPanels();
+
+        FilterPanel panelAdded;
+        int noOfPanelsToBeGenerated = Math.min(MAX_WORK_ALLOCATION_PANELS, userList.size());
+
+        List<String> filterStringsForPanels = generateListOfFilterStringsFromTurboUsers(
+                userList.subList(0, noOfPanelsToBeGenerated), "assignee:%s sort:milestone,status");
+        List<String> assigneeNamesForPanelNames = getNameListFromTurboUsers(userList.subList(0, noOfPanelsToBeGenerated));
+
+        for (int i = 0; i < noOfPanelsToBeGenerated; i++){
+            panelAdded = panelControl.addPanelAt(i);
+            panelAdded.setPanelName("Work allocated to " + assigneeNamesForPanelNames.get(i));
+            panelAdded.setFilterByString(filterStringsForPanels.get(i));
+        }
+
+        panelControl.selectFirstPanel();
+
+        String boardName = Utility.generateName(WORK_ALLOCATION,
+                prefs.getAllBoards().keySet().stream().collect(Collectors.toList()));
+
+        triggerBoardSaveEventSequence(boardName);
+
+        DialogMessage.showInformationDialog("Auto-create Board - " + WORK_ALLOCATION,
+                WORK_ALLOCATION + " board has been created and loaded.\n\n" +
+                        "It is saved under the name \"" + boardName + "\".");
+    }
+
+    private List<String> getNameListFromTurboUsers(List<TurboUser> userList) {
+        List<String> userNames = new ArrayList<>();
+        TurboUser currUser;
+        String realName, loginName;
+
+        for (int i = 0; i < userList.size(); i++) {
+            currUser = userList.get(i);
+            realName = currUser.getRealName();
+            loginName = currUser.getLoginName();
+
+            userNames.add((!realName.isEmpty() ? realName : loginName));
```

Comment:
The conditional logic in this line should be pushed to `TurboUser` class? e.g. `getFriendlierName()`